### PR TITLE
feat: make dag-pb spec compliant

### LINKED ISSUE #168: Replace `prost` with `quick-protobuf`
Currently rust-libp2p and iroh are working on getting rid of the need for `protoc` in its builds. The current plan is to switch to [`quick-protobuf`](https://github.com/tafia/quick-protobuf). 

For iroh it would be very useful if libipld switched too, as we otherwise need to avoid parts of libipld.

Refs:

- https://github.com/libp2p/rust-libp2p/issues/3024
- https://github.com/n0-computer/beetle/issues/67

### PR DESCRIPTION
**Pull request by dignifiedquire**
Closes #168, #139

Depends on 

- [x] https://github.com/multiformats/rust-multihash/pull/252
- [x] https://github.com/multiformats/rust-cid/pull/129

### CODE REVIEW

File: dag-pb/src/mod.rs [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Automatically generated mod.rs
+pub mod dag_pb;
```
Comment (vmx) [inline dag-pb/src/mod.rs]: What is this needed for?

File: dag-pb/src/codec.rs [hunk 2]
```diff
@@ -1,82 +1,65 @@
...
 use core::convert::{TryFrom, TryInto};
+use std::borrow::Cow;
+use std::collections::BTreeMap;
+
 use libipld_core::cid::Cid;
 use libipld_core::error::{Result, TypeError, TypeErrorType};
 use libipld_core::ipld::Ipld;
-use prost::bytes::{Buf, Bytes};
-use std::collections::BTreeMap;
+use quick_protobuf::sizeofs::*;
```
Comment (vmx) [inline dag-pb/src/codec.rs]:
```suggestion
use quick_protobuf::sizeofs::{sizeof_len, sizeof_varint};
```

**Review by vmx — COMMENTED** [review]
It's so nice to see this PR finally happening :)

File: dag-pb/src/mod.rs (hunk 1 above)
Comment (dignifiedquire) [reply dag-pb/src/mod.rs]: it is not 😅

**Comment by dignifiedquire** [comment]
@vmx this should be pretty ready now

**Comment by vmx** [comment]
I dared to merge my "less globs import" change, in return, I made Clippy even happier.

**Comment by vmx** [comment]
I have a look at the outstanding failures.

**Review by vmx — APPROVED** [review]
Thank you so much! Finally it's there.

Also the test fixtures at https://github.com/ipld/codec-fixtures pass.

**Pull request merged**